dev

demAirwalker opened the pull request.

demAirwalker committed:
Рефактор класса Lion с использованием инъекции зависимости.

Написание автотестов для классов Feline, Cat и Lion.

demAirwalker committed:
Рефактор класса Lion с использованием инъекции зависимости.

Написание автотестов для классов Feline, Cat и Lion.

nikolay-977 commented on .idea/compiler.xml:
```diff
@@ -0,0 +1,13 @@
+  <component name="CompilerConfiguration">
+    <annotationProcessing>
+      <profile name="Maven default annotation processors profile" enabled="true">
+        <sourceOutputDir name="target/generated-sources/annotations" />
```
Необходимо исправить: в ПР-е не должно быть лишних файлов (.DS_Store, .idea, .vscode/, build, target за исключением файлов отчета). Это локальные файлы. Загружать локальные файлы на удаленный репозиторий не принято: это считается плохой практикой. Они не несут полезной информации, но занимают место. Файлы необходимо удалить из пул-реквеста.

nikolay-977 commented on src/test/java/CatTest.java:
```diff
@@ -0,0 +1,78 @@
+
+    @Test
+    public void testGetFood() throws Exception {
+        if (expectedException == null) {
```
Можно улучшить: Не следует использовать конструкцию if-else при написании тестов. Это затрудняет читаемость кода и может вызвать путаницу в понимании. Лучше реализовать 2 отдельных кейса

nikolay-977 commented on src/test/java/FelineTest.java:
```diff
@@ -0,0 +1,88 @@
+
+    @Test
+    public void testEatMeat() throws Exception {
+        if (expectedFood != null) {
```
Можно улучшить: Не следует использовать конструкцию if-else при написании тестов. Это затрудняет читаемость кода и может вызвать путаницу в понимании. Лучше реализовать 2 отдельных кейса

nikolay-977 reviewed (commented):
Привет, Денис!

Работа выполнена хорошо, применяется инъекция зависимости. Наименование переменных, методов и классов  в соответствии с правилами нейминга. Код хорошо читаем, применяется наследование, но есть моменты, которые можно улучшить и критичные замечания, которые необходимо исправить. Буду ждать следующей итерации)

demAirwalker committed:
Рефактор кейсов с конструкцией if-else

Удаление лишних файлов из коммита

У меня только ни в какую не получается получить отчёт Jacoco, всегда какая-то ошибка :(

demAirwalker committed:
Рефактор кейсов с конструкцией if-else

Удаление лишних файлов из коммита

У меня только ни в какую не получается получить отчёт Jacoco, всегда какая-то ошибка :(

demAirwalker committed:
Рефактор кейсов с конструкцией if-else

Удаление лишних файлов из коммита

У меня только ни в какую не получается получить отчёт Jacoco, всегда какая-то ошибка, поэтому приложил .dumpstream :(
